### ISSUE TITLE
Reuse JSON transaction test vectors from Bitcoin Core

### ISSUE THREAD

**Issue by tphyahoo** [jprupp/haskoin-core#249]
context: #214 .  Note that doing this would provide bip65 testing as a side effect.  Perhaps it would catch other problems as well.  Non-passing tests could be filtered out as temporary measure to get travis to pass, while bip65 is included in haskoin.  Also, I think haskoin is not currently including tx_invalid.json tests (just tx_valid) and I obviously adding these in would further improve test coverage at little added cost.

The only format difference I see between the inlined data in haskoin 

https://github.com/haskoin/haskoin/blob/master/haskoin-core/tests/Network/Haskoin/Transaction/Units.hs

and the json in satoshi repo in

https://github.com/bitcoin/bitcoin/blob/master/src/test/data/tx_valid.json

1) comments
2) prevout scriptPubKey in inputs is represented slightly differently

tx_valid.json:

```
1 0x41 0x04cc71eb30d653c0c3163990c47b976f3fb3f37cccdcbedb169a1dfef58bbfbfaff7d8a473e7e2e6d317b87bafe8bde97e3cf8f065dec022b51d11fcdd0d348ac4 0x41 0x0461cbdcc5409fb4b4d42b51d33381354d80e550078cb532a34bfa2fcfdeb7d76519aecc62770f5b0e4ef8551946d8a540911abe3e7854a26f39f58b25c15342af 2 OP_CHECKMULTISIG"
```

haskoin:

```
514104cc71eb30d653c0c3163990c47b976f3fb3f37cccdcbedb169a1dfef58bbfbfaff7d8a473e7e2e6d317b87bafe8bde97e3cf8f065dec022b51d11fcdd0d348ac4410461cbdcc5409fb4b4d42b51d33381354d80e550078cb532a34bfa2fcfdeb7d76519aecc62770f5b0e4ef8551946d8a540911abe3e7854a26f39f58b25c15342af52ae
```

Same thing, except some of the op codes are spelled out in the json.  So if we have a function to translate the first string to the second (and I am sure we do, I just don't know where yet), we can almost recycle existing testing code unchanged.

**Comment by jprupp** [comment]
If you want to do it be my guest. A string of opcodes in the form "OP_CODE1 OP_CODE2" can be converted to the form [OP_CODE1, OP_CODE2] and then parsed with the `read` command in Haskell into a normal list. In fact. it is possible to create data type that can parse that string directly quite easily, since the `Show`/`Read` instances for opcodes can be reused for that.

**Comment by tphyahoo** [comment]
More guidance would be helpful.

Once we have a string that looks like

```
" [OP_CODE1, OP_CODE2]"
```

what data type should this be read/parsed into? 

Looking at the haddock for Network.Haskoin.Script I guessed perhaps 

```
data Script =
    Script {
             -- | List of script operators defining this script
             scriptOps :: ![ScriptOp]
```

but my naive first guess doesn't seem to work.  I confess I'm a little uncertain how to proceed.  Is this the right data type (and I just need to work on parser / understanding the problem better), or should I be aiming for a different type?

```
-- first attempt
t1 :: [ScriptOp]
t1 = map read . words $ tx_validJsonString
tx_validJsonString = "1 0x41 0x04cc71eb30d653c0c3163990c47b976f3fb3f37cccdcbedb169a1dfef58bbfbfaff7d8a473e7e2e6d317b87bafe8bde97e3cf8f065dec022b51d11fcdd0d348ac4 0x41 0x0461cbdcc5409fb4b4d42b51d33381354d80e550078cb532a34bfa2fcfdeb7d76519aecc62770f5b0e4ef8551946d8a540911abe3e7854a26f39f58b25c15342af 2 OP_CHECKMULTISIG"

```

yields

```
t1
[*** Exception: Prelude.read: no parse
```

**Comment by tphyahoo** [comment]
Maybe this is the right track after all, I just need to work on parser.

If string starts "OP_1" instead of "1" it parses that bit, and chokes soon after, but at least it can parse something.

I guess this is the right approach then?

**Comment by jprupp** [comment]
Hmm, it seems you will need to do some manual work to make that parse.

**Comment by jprupp** [comment]
You would need to work on a parser as you said.

**Comment by tphyahoo** [comment]
Fixed by #255.

**Comment by jprupp** [comment]
Fix has been merged into master.
